modules/hal_afbr/platform_s2pi.c

```diff
@@ -0,0 +1,450 @@
+				    txData, frameSize, NULL);
+	}
+	xfer_sqe->flags |= RTIO_SQE_CHAINED;
+
+	rtio_sqe_prep_callback_no_cqe(cb_sqe, S2PI_complete_callback, data, NULL);
+
+	rtio_submit(ctx, 0);
+
+	return STATUS_OK;
+}
+
+status_t S2PI_Abort(s2pi_slave_t slave)
+{
+	struct afbr_s50_platform_data *data;
+	int err;
+
+	err = afbr_s50_platform_get_by_id(slave, &data);
+	CHECKIF(err) {
+		return ERROR_S2PI_INVALID_SLAVE;
+	}
+
+	(void)atomic_set(&data->s2pi.rtio.state, ERROR_ABORTED);
+
+	S2PI_complete_callback(data->s2pi.rtio.ctx, NULL, data);
+
+	return STATUS_OK;
+}
+
+status_t S2PI_SetIrqCallback(s2pi_slave_t slave,
+			     s2pi_irq_callback_t callback,
+			     void *callbackData)
+{
+	struct afbr_s50_platform_data *data;
+	int err;
+
+	err = afbr_s50_platform_get_by_id(slave, &data);
+	CHECKIF(err) {
+		return ERROR_S2PI_INVALID_SLAVE;
+	}
+
+	data->s2pi.irq.handler = callback;
+	data->s2pi.irq.data = callbackData;
+
+	if (data->s2pi.irq.handler) {
+		err = gpio_pin_interrupt_configure_dt(data->s2pi.gpio.irq, GPIO_INT_EDGE_TO_ACTIVE);
+		CHECKIF(err) {
+			LOG_ERR("Failed to enable IRQ GPIO: %d", err);
+			return ERROR_FAIL;
+		}
+	} else {
+		err = gpio_pin_interrupt_configure_dt(data->s2pi.gpio.irq, GPIO_INT_DISABLE);
+		CHECKIF(err) {
+			LOG_ERR("Failed to disable IRQ GPIO: %d", err);
+			return ERROR_FAIL;
+		}
+	}
+
+	return STATUS_OK;
+}
+
+uint32_t S2PI_ReadIrqPin(s2pi_slave_t slave)
+{
+	struct afbr_s50_platform_data *data;
+	int err;
+
+	err = afbr_s50_platform_get_by_id(slave, &data);
+	CHECKIF(err) {
+		LOG_ERR("Error getting platform data: %d", err);
+		return 0;
+	}
+
+	const struct gpio_dt_spec *irq = data->s2pi.gpio.irq;
+	int value;
+
+	value =  gpio_pin_get_dt(irq);
+	CHECKIF(value < 0) {
+		LOG_ERR("Error reading IRQ pin: %d", value);
+		return 0;
+	}
+
+	return !value;
+}
+
+status_t S2PI_CycleCsPin(s2pi_slave_t slave)
+{
+	struct afbr_s50_platform_data *data;
+	int err;
+
+	err = afbr_s50_platform_get_by_id(slave, &data);
+	CHECKIF(err) {
+		LOG_ERR("Error getting platform data: %d", err);
+		return ERROR_S2PI_INVALID_SLAVE;
+	}
+
+	const struct gpio_dt_spec *cs = data->s2pi.gpio.spi.cs;
+
+	err = gpio_pin_set_dt(cs, 1);
+	CHECKIF(err) {
+		LOG_ERR("Error setting CS pin logical high: %d", err);
+		return ERROR_FAIL;
+	}
+
+	err = gpio_pin_set_dt(cs, 0);
+	CHECKIF(err) {
+		LOG_ERR("Error setting CS pin logical low: %d", err);
+		return ERROR_FAIL;
+	}
+
+	return STATUS_OK;
+}
+
+status_t S2PI_CaptureGpioControl(s2pi_slave_t slave)
+{
+	struct afbr_s50_platform_data *data;
+	int err;
+
+	err = afbr_s50_platform_get_by_id(slave, &data);
+	CHECKIF(err) {
+		LOG_ERR("Error getting platform data: %d", err);
+		return ERROR_S2PI_INVALID_SLAVE;
+	}
+
+	err = pinctrl_apply_state(data->s2pi.pincfg, PINCTRL_STATE_PRIV_START);
+	CHECKIF(err) {
+		LOG_ERR("Error applying gpio pinctrl state: %d", err);
+		return ERROR_FAIL;
+	}
+
+	err = gpio_pin_configure_dt(data->s2pi.gpio.spi.miso, GPIO_OUTPUT);
```

Comment:
shouldn't MISO be INPUT?

Reply:
> In addition to the standard SPI interface, the corresponding pins must also be accessible in GPIO mode. This is required to access the EEPROM memory of the AFBR-S50 sensor device that holds calibration parameters

Ref: https://broadcom.github.io/AFBR-S50-API/porting_guide.html

While not SPI (this is GPIO mode), I just confirmed on the reference integration MISO is still configured as an input (w/ pull-up):
https://github.com/Broadcom/AFBR-S50-API/blob/a886f463c6db5ae487fd5a3c5d9fb13545a4fc7b/Sources/Platform/NXP_MKLxxZ/driver/s2pi.c#L1033-L1043

This does not impact the current code as calibration is not performed, but I'll address it to guard us in the future. Thanks!